### PR TITLE
feat: make jitter interval configurable

### PR DESCRIPTION
**Pull request by raghunathd8**
Make jitter interval configurable. Add new config parameter: `JitterFactor` which defaults to [original](https://github.com/cloudfoundry/route-emitter/blob/main/scheduler/routebroadcastscheduler.go#L98C2-L98C2) `0.2`.

### CODE REVIEW

File: scheduler/routebroadcastscheduler.go [hunk 1]
```diff
@@ -30,6 +31,7 @@ func NewRouteBroadcastScheduler(
 	natsClient diegonats.NATSClient,
 	logger lager.Logger,
 	externalServiceName string,
+	jitter_factor float64,
```
Comment (maxmoehl) [inline scheduler/routebroadcastscheduler.go:34]: Please always use camelCase instead of snake_case in go.

File: cmd/route-emitter/main.go [hunk 2]
```diff
@@ -73,8 +73,8 @@ func main() {
 	externalChan := make(chan struct{}, 1)
 	internalChan := make(chan struct{}, 1)
 	syncer := syncer.NewSyncer(clock, time.Duration(cfg.SyncInterval), logger)
-	externalScheduler := scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, "router", externalChan)
-	internalScheduler := scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, "service-discovery", internalChan)
+	externalScheduler := scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, "router", cfg.JitterFactor, externalChan)
```
Comment (domdom82) [inline cmd/route-emitter/main.go:76]:
```suggestion
	externalScheduler := scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, "router", cfg, externalChan)
```

I'd prefer handing down the config and make it more flexible. If there are more properties to configure in the future, we won't have to change this again.

File: cmd/route-emitter/main.go [hunk 3]
```diff
@@ -73,8 +73,8 @@ func main() {
...
+	internalScheduler := scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, "service-discovery", cfg.JitterFactor, internalChan)
```
Comment (domdom82) [inline cmd/route-emitter/main.go:77]:
```suggestion
	internalScheduler := scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, "service-discovery", cfg, internalChan)
```

File: scheduler/routebroadcastscheduler.go [hunk 4]
```diff
@@ -30,6 +31,7 @@ func NewRouteBroadcastScheduler(
 	natsClient diegonats.NATSClient,
 	logger lager.Logger,
 	externalServiceName string,
+	jitterFactor float64,
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler.go:34]:
```suggestion
	cfg *config,
```

File: scheduler/routebroadcastscheduler.go [hunk 5]
```diff
@@ -21,6 +21,7 @@ type RouteBroadcastScheduler struct {
 	clock                clock.Clock
 	emitCh               chan struct{}
 	externalServiceStart chan time.Duration
+	jitterFactor         float64
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler.go:24]:
```suggestion
	cfg         *config
```

File: scheduler/routebroadcastscheduler.go [hunk 6]
```diff
@@ -40,6 +42,7 @@ func NewRouteBroadcastScheduler(
 		emitCh: emitCh,
 
 		externalServiceStart: make(chan time.Duration),
+		jitterFactor:         jitterFactor,
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler.go:45]:
```suggestion
		cfg:         cfg,
```

File: scheduler/routebroadcastscheduler.go [hunk 7]
```diff
@@ -95,7 +98,7 @@ GREET_LOOP:
 		select {
 		case registerInterval = <-s.externalServiceStart:
 			s.logger.Info("received-new-external-service-prune-interval", lager.Data{"interval": registerInterval.String()})
-			jitterInterval := randSource.Int63n(int64(0.2 * float64(registerInterval)))
+			jitterInterval := randSource.Int63n(int64(s.jitterFactor * float64(registerInterval)))
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler.go:101]:
```suggestion
			jitterInterval := randSource.Int63n(int64(s.cfg.jitterFactor * float64(registerInterval)))
```

File: scheduler/routebroadcastscheduler_test.go [hunk 8]
```diff
@@ -21,6 +21,7 @@ var _ = Describe("RouteBroadcastScheduler", func() {
 		schedulerRunner *scheduler.RouteBroadcastScheduler
 		process         ifrit.Process
 		clock           *fakeclock.FakeClock
+		jitterFactor    float64
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler_test.go:24]: user a default config here instead

File: scheduler/routebroadcastscheduler_test.go [hunk 9]
```diff
@@ -33,6 +34,8 @@ var _ = Describe("RouteBroadcastScheduler", func() {
 			BeforeEach(func() {
 				natsClient = diegonats.NewFakeClient()
 
+				jitterFactor = 0.2
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler_test.go:37]: fill the value for `jitterFactor` of the default config here

File: scheduler/routebroadcastscheduler_test.go [hunk 10]
```diff
@@ -52,7 +55,7 @@ var _ = Describe("RouteBroadcastScheduler", func() {
 
 			JustBeforeEach(func() {
 				logger := lagertest.NewTestLogger("test")
-				schedulerRunner = scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, prefix, emitCh)
+				schedulerRunner = scheduler.NewRouteBroadcastScheduler(clock, natsClient, logger, prefix, jitterFactor, emitCh)
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler_test.go:58]: pass down a pointer to the default config instead

**Review by domdom82 — CHANGES_REQUESTED** [review]

File: scheduler/routebroadcastscheduler_test.go [hunk 11]
```diff
@@ -32,6 +34,7 @@ var _ = Describe("RouteBroadcastScheduler", func() {
 		Context(prefix, func() {
 			BeforeEach(func() {
 				natsClient = diegonats.NewFakeClient()
+				cfg.JitterFactor = 0.2
```
Comment (domdom82) [inline scheduler/routebroadcastscheduler_test.go:37]: this won't work. you've initialized`cfg` as a pointer to config.RouteEmitterConfig in line 25 but never gave it any value to point to. so this test will throw a nil ref panic. please run your tests first before requesting a review.

I've added a suggestion that fixes the issue.
```suggestion
				cfg = &config.RouteEmitterConfig{}
				cfg.JitterFactor = 0.2
```

**Review by domdom82 — CHANGES_REQUESTED** [review]

**Review by domdom82 — APPROVED** [review]

**Review by ameowlia — APPROVED** [review]

**Pull request merged**